refactor(authenticator)!: use Amplify JS auto sign in (#2642)
aws-amplify/amplify-ui: docs/src/pages/[platform]/getting-started/migration/migration.vue.mdx

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation
For Vue.js 2 users please continue to use our legacy library found [here](https://github.com/aws-amplify/amplify-ui/tree/legacy/legacy/amplify-ui-vue).

> Install the 3.x version of the `@aws-amplify/ui-vue` library.

Reviewer: ```suggestion
Install the 3.x version of the `@aws-amplify/ui-vue` and the 5.x version of the `aws-amplify` libraries.
```

Author: Done

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation
For Vue.js 2 users please continue to use our legacy libary found [here](https://github.com/aws-amplify/amplify-ui/tree/legacy/legacy/amplify-ui-vue).

Install the 3.x version of the `@aws-amplify/ui-vue` library.

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-vue` major version `3.0`:

> #### 1. Update any override function calls

Reviewer 2: ```suggestion
#### 1. Update any function override calls
```

Author: +1

> Update the override function calls with the updated function signatures. Rather then returning an `Auth.*` object you must now return the function callback passed in the `handle*` signature.

Reviewer 2: ```suggestion
Update the function override calls with the updated function signatures. Rather than returning an `Auth.*` object you must now return the function callback passed in the `handle*` signature.
```

```diff
- async handleSignUp(formData) {
+ async handleSignUp(formData, signUp) {
  let { username, password, attributes } = formData;
  // custom username
  username = username.toLowerCase();
  attributes.email = attributes.email.toLowerCase();
-  return Auth.signUp({
-    username,
-    password,
-    attributes,
-  })
+  return signUp(username, password, attributes)

> }

Author: Done

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation

> For Vue.js 2 users please continue to use our legacy libary found [here](https://github.com/aws-amplify/amplify-ui/tree/legacy/legacy/amplify-ui-vue).

Reviewer 2: ```suggestion
For Vue.js 2 users please continue to use our legacy library found [here](https://github.com/aws-amplify/amplify-ui/tree/legacy/legacy/amplify-ui-vue).
```

Author: +1

Reviewer: Looks like the typo is still in there

Install the 3.x version of the `@aws-amplify/ui-vue` library.

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-vue` major version `3.0`:

> #### 1. Update the handleSignUp function override call

Reviewer 3: ```suggestion
#### 1. Update custom handleSignUp function override calls
```

Author: Done

> Update the override function call to include the `autSignIn` set to enabled. This must be included if you are overriding `Auth.signUp`.

Reviewer 3: Should the conditional of when this patch needs to be applied move to the front?
```suggestion
If you are overriding `Auth.signUp`, update the override function call to include the `autoSignIn` option set to enabled. If this change is not made, customers will not be automatically signed in on signup.
```

Author: Updated

import { Tabs, TabItem } from '@aws-amplify/ui-react';
import { Alert } from '@aws-amplify/ui-react';

## 2.x to 3.x
### Installation
For Vue.js 2 users please continue to use our legacy libary found [here](https://github.com/aws-amplify/amplify-ui/tree/legacy/legacy/amplify-ui-vue).

Install the 3.x version of the `@aws-amplify/ui-vue` library.

<Tabs>
<TabItem title="npm">

```shell
npm install aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
<TabItem title="yarn">
```shell
yarn add aws-amplify @aws-amplify/ui-vue@3.x
```

</TabItem>
</Tabs>

### Update and usage

The following breaking changes were introduced in `@aws-amplify/ui-vue` major version `3.0`:

> #### Automatic signin on signup logic moved to `aws-amplify`. 

Reviewer: Similar comment to the react migration guide. Let's add updating the libraries version to 5 above this heading, add a '1.' to this heading and remove the L5 heading below.

Author: Fixed
